frontend/javascripts/maintenance_banner.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  MaintenanceInfo,
+  listCurrentAndUpcomingMaintenances,
+  updateNovelUserExperienceInfos,
+} from "admin/admin_rest_api";
+import { Alert } from "antd";
+import FormattedDate from "components/formatted_date";
+import { useInterval } from "libs/react_helpers";
+import { setActiveUserAction } from "oxalis/model/actions/user_actions";
+import { Store } from "oxalis/singletons";
+import React, { useState } from "react";
+
+export function MaintenanceBanner() {
+  const { activeUser } = Store.getState();
+  const [currentAndUpcomingMaintenances, setcurrentAndUpcomingMaintenances] = useState<
+    Array<MaintenanceInfo>
+  >([]);
+  const [position, setPosition] = useState<Object>({ top: 48 });
+  const [isTop, setIsTop] = useState(true);
+  useInterval(
+    async () => {
+      if (activeUser == null) return;
+      setcurrentAndUpcomingMaintenances(await listCurrentAndUpcomingMaintenances());
+    },
+    10000, //TODO find value for production
+  );
+  if (activeUser == null) return <></>;
+  const activeUsersLatestAcknowledgedMaintenance =
+    activeUser?.novelUserExperienceInfos.latestAcknowledgedMaintenanceInfo;
+
+  const saveUserClosedMaintenanceInfo = (closestUpcomingMaintenance: MaintenanceInfo) => {
+    if (activeUser == null) return;
+    const [nextMaintenanceAcknowledged] = updateNovelUserExperienceInfos(activeUser, {
+      latestAcknowledgedMaintenanceInfo: closestUpcomingMaintenance.id,
+    });
+    Store.dispatch(setActiveUserAction(nextMaintenanceAcknowledged));
+  };
+
+  const toggleTopOrBottomPosition = () => {
+    setPosition(isTop ? { top: "48px" } : { bottom: "20px" });
+    setIsTop(!isTop);
+  };
+
+  const getClosestUpcomingMaintenanceBanner = () => {
+    const currentTime = Date.now();
+    const closestUpcomingMaintenance = currentAndUpcomingMaintenances
+      ?.filter((maintenance) => maintenance.startTime > currentTime)
+      .sort((a, b) => a.startTime - b.startTime)[0];
+    //if (closestUpcomingMaintenance == null || activeUsersLatestAcknowledgedMaintenance === closestUpcomingMaintenance.id) return; //TODO use this after testing and remove next 3 lines
+    if (closestUpcomingMaintenance == null) return;
+    if (activeUsersLatestAcknowledgedMaintenance === closestUpcomingMaintenance.id)
+      return <Alert message="Latest maintenance already acknowledged" type="success" />;
```

Comment:
this is probably only used for testing?

Reply:
yes, hence the comment :) can I already delete this?

Reply:
yes, I'd say so :)